Document all language REPLs and ANSI support (#2442)
pygments/pygments: doc/docs/terminal-sessions.rst

=================
Terminal sessions
=================

Pygments support the parsing and highlighting of terminal sessions, like
command-line shells, interactive consoles and language `REPL
<https://en.wikipedia.org/wiki/Read–eval–print_loop>`_.

> They are typiccaly command lines or code, mixed with generic output.

Reviewer: ```suggestion
They are typically command lines or code, mixed with generic output.
```

Examples for each can be found in the :doc:`lexer <lexers>` documentation.

Operating system shells
-----------------------

These lexers are expecting a prompt to identify user input. So to highlight a
shell session, prefix your code snippet with a specially formatted prompt.

> They are typiccaly named ``<shell> Session``.

Reviewer: ```suggestion
They are typically named ``<shell> Session``.
```

Interactive consoles
--------------------

Similarly to systems shells, Pygments recognize a variety of interactive
language sessions.

> Their IDs are typiccaly follow the ``<language>-console`` or

Reviewer: ```suggestion
Their IDs typically follow the ``<language>-console`` or
```

``<language>-repl`` pattern.

Generic output
--------------

To display standalone terminal output and keep styling consistent, you can use
the generic ``output`` lexer.

ANSI rendering
--------------

In all the lexers above, the command results are parsed as generic output.

> Which means they are rendered as-is, without any tentative of rendering
> the styling carried on by ANSI codes.

Reviewer: ```suggestion
Which means they are rendered as-is, without any styling applied, for example by ANSI codes.
```

     % ls -lh
     > ls -lh

* **MSDOS Session** (``doscon``):

  .. code-block:: doscon

     [any]> dir
     > dir
     More? dir

* **PowerShell Session** (``ps1con``, ``pwsh-session``):

  .. code-block:: ps1con

     PS[any]> Get-ChildItem
     PS> Get-ChildItem
     >> Get-ChildItem

* **Tcsh Session** (``tcshcon``):

  .. code-block:: tcshcon

     (any)> ls -lh
     ? ls -lh

Interactive consoles
--------------------

Similarly to systems shells, Pygments recognize a variety of interactive

> language sessions:

Author: OK, I just moved all the examples to the docstrings.
